File: LeetCode/Buy_and_Sell_Stocks.js
let maxProfit = function(prices){
	let max_profit = 0;
	let min_prices = prices[0];

	for(let i = 0; i<prices.length; i++){
		if(min_prices > prices[i])
			min_prices = prices[i];

		if(max_profit < prices[i] - min_prices)
			max_profit = prices[i] - min_prices;
	}
	return max_profit;
}

let readline = require('readline');
let read = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

read.question("", function(array){
	array = array.split(' ').map((value) => {
		return Number(value);
	});
	let result = maxProfit(array);
	console.log(result);
	read.close();
})

read.close('', function(){
	process.exit();
})

File: LeetCode/Buy_and_Sell_Stocks.py
def maxProfit(prices):
	max_profit = 0;
	min_prices = prices[0]

	for i in range(len(prices)):
		min_prices = min(min_prices, prices[i])
		max_profit = max(max_profit, prices[i] - min_prices)

	return max_profit 


prices = list(map(int, input().split()))
result = maxProfit(prices)
print(result)